core-httpclient-impl/src/main/java/com/optimizely/ab/odp/DefaultODPApiManager.java

```diff
@@ -205,7 +205,7 @@ public Integer sendEvents(String apiKey, String apiEndpoint, List<ODPEvent> even
             response = httpClient.execute(request);
         } catch (IOException e) {
             logger.error("Error retrieving response from event request", e);
-            return null;
+            return 0;
```

Comment:
Either null or 0 is fine with me. My question was that how we can tell if this failure should be retried or not at the upper level. Do you plan to retry all status of 0 and 50x?

Reply:
yes! I will document it on the method too.
1. null means there was a non recoverable error and no retry is needed.
2. 0 means an unexpected error and you should retry
3. Any other status can be handled as needed but my idea is to return on 500+